crates/wasmtime/src/compile.rs

```diff
@@ -493,7 +508,26 @@ impl<'a> CompileInputs<'a> {
         let compiler = engine.compiler();
 
         // Compile each individual input in parallel.
-        let raw_outputs = engine.run_maybe_parallel(self.inputs, |f| f(compiler))?;
+        let mut raw_outputs = engine.run_maybe_parallel(self.inputs, |f| f(compiler))?;
+
+        // Now that all functions have been compiled see if any wasmtime-builtin
+        // functions are necessary. If so those need to be collected and then
+        // those trampolines additionally need to be compiled.
+        let mut builtins = BTreeSet::new();
+        for output in raw_outputs.iter_mut() {
+            builtins.extend(output.builtins.drain(..));
+        }
+        let builtins = builtins.into_iter().collect::<Vec<_>>();
+        raw_outputs.extend(engine.run_maybe_parallel(builtins, |i| -> Result<_> {
+            let symbol = format!("wasmtime_builtin{}", i.index());
```

Comment:
We could probably define a `fn builtin_name(builtin: BuiltinFunctionIndex) -> &'static str` method via the macros to get better names here, which could be useful in debugging/profiling.